app/controllers/email/VerifyPasscodeController.scala

```diff
@@ -153,27 +206,72 @@ class VerifyPasscodeController @Inject()(emailVerificationService: EmailVerifica
   }
 
   def contactPrefSendVerification: Action[AnyContent] = (contactPreferencePredicate andThen
-                                                         paperPrefPredicate) { implicit user =>
+                                                         paperPrefPredicate andThen
+                                                         inFlightContactPrefPredicate).async { implicit user =>
+    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(user.headers, Some(user.session))
 
-    if (appConfig.features.emailPinVerificationEnabled()) {
+    if (appConfig.features.emailPinVerificationEnabled() && appConfig.features.letterToConfirmedEmailEnabled()) {
       extractSessionEmail match {
-        case Some(_) => Ok("") //TODO
-        case _ => Redirect(controllers.contactPreference.routes.ContactPreferenceRedirectController.redirect())
+        case Some(email) => emailVerificationService.isEmailVerified(email).flatMap {
```

Comment:
This `contactPrefSendVerification` action needs some rework. This should be almost identical to the `emailSendVerification` action above. They are functionally almost the same but just split across two different journeys. For example compare `emailShow` and `contactPrefShow`. You just need to call the `emailVerificationService.createPasscodeRequest` function and that will give you the info you need to decide where the user is going.